README.md

```diff
@@ -163,17 +164,20 @@ If the config-helper doesn't provide the level of customization you need for you
 | SYSTEM_TAG | Tag which will be applied to all MLSpace resources created with the AWS account to which MLSpace is deployed | `MLSpace` |
 | IAM_RESOURCE_PREFIX | Value preprended to MLSpace dynamic roles and policies when `MANAGE_IAM_ROLES` is set to `true` | `MLSpace` |
 | MANAGE_IAM_ROLES | This setting determines whether or not MLSpace will dynamically create unique roles per project/user combinations. | `true` |
 | NOTIFICATION_DISTRO | Optional email distribution list which will be notified when <TBD> | - |
 | EXISTING_VPC_NAME | If MLSpace is being deployed into an existing VPC this should be the name of that VPC (must also set `EXISTING_VPC_ID`) | - |
 | EXISTING_VPC_ID | If MLSpace is being deployed into an existing VPC this should be the id of that VPC (must also set `EXISTING_VPC_NAME`) | - |
 | EXISTING_VPC_DEFAULT_SECURITY_GROUP | If MLSpace is being deployed into an existing VPC this should be the default security group of that VPC | - |
-| APP_ROLE_ARN | ARN of an existing IAM role to use for executing the MLSpace lambdas. This value must be set to an existing role because the default CDK deployment will not create one. | - |
-| NOTEBOOK_ROLE_ARN | ARN of an existing IAM role to associate with all notebooks created in MLSpace. If using dynamic roles based on project/user combinations the specific combination role will be used instead. This value must be set to an existing role because the default CDK deployment will not create one. | - |
-| S3_READER_ROLE_ARN | ARN of an existing IAM role to use for reading from the static website S3 bucket. If not specified a new role with the correct privileges will be created | - |
+| APP_ROLE_ARN | Arn of an existing IAM role to use for executing the MLSpace lambdas. This value must be set to an existing role because the default CDK deployment will not create one. | - |
+| NOTEBOOK_ROLE_ARN | Arn of an existing IAM role to associate with all notebooks created in MLSpace. If using dynamic roles based on project/user combinations the specific combination role will be used instead. This value must be set to an existing role because the default CDK deployment will not create one. | - |
+| SYSTEM_ROLE_ARN | Arn of an existing IAM role to use for executing the MLSpace system lambdas. System lambdas are responsible for maintaining the MLSpace system by cleaning up resources when a project is suspended or deleted, when a user is suspended, or when services are activated/deactivated. This value must be set to an existing role because the default CDK deployment will not create one. | - |
+| ENDPOINT_CONFIG_INSTANCE_CONSTRAINT_POLICY_ARN | ARN for policy constraining the instance size that can be used when creating Endpoint configurations from a notebook. | - |
```

Comment:
@dustins correct me if I wrong, but it was worded like this because the authorizer + the UI simply not presenting these instance types will handle the UI usecase, whereas these IAM policies are here to stop notebook users from using these services right?

Reply:
That's correct. Those policies are specific to notebooks.

But thinking about it I guess _technically_ a user could submit a job with the incorrect instance type if they bypassed the UI right? Is there any downside to attaching these to the app role too? (other than using 2 more policy slots on that role)

Reply:
I think we should move forward with merging these changes, if I make changes to apply these to the app role I'll fix the docs.

Reply:
Using up 2 more slots is potentially a significant drawback if we ever need those slots in the future. How would someone bypass the UI without using a notebook? You're thinking they could send a curl command in the browser terminal or something? 

At any rate these descriptions are indeed accurate as they're written today, so I'll resolve this thread.

Reply:
There is a semi-easy hack for this via the UI then. With dev tools I think you could modify a value to whatever instance you want and launch it since we aren't doing any validation against the list of allowed instances on the front or backend.

Not sure anyone would do this, but I think it is possible.

I would ask the PM about this, because if I was a customer this would be confusing. It does limit the UI as well, so saying it just affects notebooks still feels misleading.

Reply:
Discussed in meeting; will leave the description as-is since it's currently true, but we're planning to update this so this policy is in the app role as well. We will update the documentation accordingly when the application code is udpated.